### ISSUE TITLE
Simplify pbt_deep_dup_build conditional checks

### ISSUE THREAD

**Issue by danielpclark** [danielpclark/PolyBelongsTo#7]
This block of code:

```ruby
if child.respond_to?(:build)
  child.each do |spawn|
    if core.respond_to?(:build)
      core.each do |subscore|
        subscore.pbt_deep_dup_build(spawn)
      end
    else
      core.pbt_deep_dup_build(spawn)
    end
  end
else
  if core.respond_to?(:build)
    core.each do |subcore|
      subcore.pbt_deep_dup_build(child)
    end
  else
    core.pbt_deep_dup_build(child)
  end
end
```
Needs to be refactored into something much simpler.  It works, but it shouldn't have to be this excessive.

**Comment by danielpclark** [comment]
This may be achieved if we create a new method to replace PolyBelongsTo::Pbt::CollectionProxy with something like PolyBelongsTo::Pbt::ForcedProxy

Currently CollectionProxy returns a proxy for has_many and an instance for has_one.  It should be left as is for backwards compatibility.  The new method should take single has_one objects and place them inside a collection proxy regardless of the has_one relationship.

I'm not committed to the name ForcedProxy, I'm open to suggestions.

**Comment by danielpclark** [comment]
I'm most of the way through implementing this.  Just a heads up.

**Comment by danielpclark** [comment]
The methods are implemented: https://github.com/danielpclark/PolyBelongsTo/tree/faked-collection

Both **AsCollectionProxy** and **FakedCollection**.  I can't seem to focus enough to figure how to use it in the refactoring of method **pbt_deep_dup_build**.  So I'll give some time for this to simmer before it gets done.

**Comment by danielpclark** [comment]
Merged with pull request #8
